### DISCUSSION TITLE
how to evolve enum to add experimental cases?

### DISCUSSION THREAD

**Discussion by bishabosha** [scala/scala3#17815] [Feature Requests]
## Compiler version

3.1.0

## Minimized code

```Scala
enum Foo:
  @annotation.experimental case A
```

## Output

```scala
-- Error:
2 |  @annotation.experimental case A
  |  ^
  |value A is marked @experimental and therefore may only be used in an experimental scope.
```

## Expectation
The error makes sense, as `values`, `valueOf`, `fromOrdinal` etc reference `A`, but it does mean we can not "evolve" our enum to add new cases without making every method experimental.

Or perhaps those methods should all automatically become experimental as soon as an experimental case is added? (not good for downstream users of a library)

**Comment by odersky** [comment]
If we really care about this (I personally do not see a strong usecase), this should be a feature request.
